packages/workflow/src/NodeReferenceParserUtils.ts

```diff
@@ -69,3 +119,462 @@ export function applyAccessPatterns(expression: string, previousName: string, ne
+				};
+			}
+		}
+	}
+
+	// This covers specific metadata functions available on nodes
+	const itemAccessorMatch = ITEM_ACCESSORS.flatMap((x) => (x === parts[0] ? x : []))[0];
+	if (itemAccessorMatch !== undefined) {
+		return {
+			nodeNameInExpression,
+			originalExpression: `${exprStart}.${parts[0]}`, // $('abc').isExecuted
+			replacementPrefix: `$('${startNodeName}').first().json`, //  $('Start').first()
+			replacementName: `${nodeNamePlainJs}_${parts[0]}`, // nodeName_isExecuted
+		};
+	}
+
+	// If we end up here it means that:
+	// - we have a complex `itemMatching(<expr>)` case, or
+	// - the expression should be invalid, or
+	// - a new function was added that we're not aware of.
+	//
+	// In these cases let's just not touch it and keep it as is
+	return null;
+}
+
+// find `$('NodeName').item.json.path.to.x` in `{{ $('NodeName').item.json.path.to.x[someFunction()] }}`
+function extractExpressionCandidate(expression: string, startIndex: number, endIndex: number) {
+	const firstPartException = ITEM_TO_DATA_ACCESSORS.map((x) =>
+		x.exec(expression.slice(endIndex)),
+	).filter((x) => x !== null);
+
+	// Note that by choosing match 0 we use `itemMatching` matches over `item`
+	// matches by relying on the order in ITEM_TO_DATA_ACCESSORS
+	const after_accessor_idx = endIndex + (firstPartException[0]?.[0].length ?? -1) + 1;
+	const after_accessor = expression.slice(after_accessor_idx);
+	const firstInvalidCharMatch = INVALID_JS_DOT_PATH.exec(after_accessor);
+
+	// we should at least find the }} closing the JS expressions in valid cases
+	if (!firstInvalidCharMatch) return null;
+
+	return expression.slice(startIndex, after_accessor_idx + firstInvalidCharMatch.index);
+}
+
+// Parse a given regex accessor match (e.g. `$('nodeName')`, `$node['nodeName']`)
+// and extract a potential ExpressionMapping
+function parseCandidateMatch(
+	match: RegExpExecArray,
+	expression: string,
+	nodeNames: string[],
+	startNodeName: string,
+): ExpressionMapping | null {
+	const startIndex = match.index;
+	const endIndex = startIndex + match[0].length + 1;
+	// this works because all access patterns define match groups
+	// [fullMatch, "$('", "nodeName", "')"]
+	const nodeNameInExpression = match[2];
+
+	// This should be invalid in theory, since the regex matches should only act
+	// on known node names
+	if (!nodeNames.includes(nodeNameInExpression)) return null;
+
+	const candidate = extractExpressionCandidate(expression, startIndex, endIndex);
+	if (candidate === null) return null;
+	return parseExpressionMapping(
+		candidate,
+		nodeNameInExpression,
+		convertToUniqueJsDotName(nodeNameInExpression, nodeNames),
+		startNodeName,
+	);
+}
+
+// Handle matches of form `$json.path.to.value`, which is necessary for the selection input node
+function parse$jsonMatch(match: RegExpExecArray, expression: string, startNodeName: string) {
+	const candidate = extractExpressionCandidate(expression, match.index, match[0].length);
+	if (candidate === null) return;
```

Comment:
minor: this should probably also just return `null` to keep the return value of this function consistent `ExpressionMapping | null` like on most other functions here instead of `ExpressionMapping | null | undefined`